src/tester.rs

```diff
@@ -301,25 +301,39 @@ impl<T: Testable,
             self_: fn($($name),*) -> T,
             a: ($($name,)*),
         ) -> Option<TestResult> {
-            for t in a.shrink() {
-                let ($($name,)*) = t.clone();
+            let mut values = a.shrink().collect::<Vec<_>>();
```

Comment:
Good catch, and I can resolve that easily - the realisation of the iterator was only necessary back when I thought this was doing a full depth search (with backtracking). Since we don't do that, we can just use an iterator in the while loop, and reassign it when we descend.